helix-core/src/main/java/org/apache/helix/util/HelixUtil.java

```diff
@@ -201,23 +201,42 @@ public static String serializeByComma(List<String> objects) {
    */
   public static Map<String, String> computeIdealMapping(List<String> preferenceList,
       StateModelDefinition stateModelDef, Set<String> liveAndEnabled) {
+    return computeIdealMapping(preferenceList, stateModelDef, liveAndEnabled,
+        Collections.emptySet());
+  }
+
+  /**
+   * compute the ideal mapping for resource in Full-Auto and Semi-Auto based on its preference list
+   */
+  public static Map<String, String> computeIdealMapping(List<String> preferenceList,
+      StateModelDefinition stateModelDef, Set<String> liveInstanceSet,
+      Set<String> disabledInstancesForPartition) {
     Map<String, String> idealStateMap = new HashMap<String, String>();
 
     if (preferenceList == null) {
       return idealStateMap;
     }
 
+    for (String instance : preferenceList) {
```

Comment:
If you fixed here, it will break the rule of StrictMatchVerifier for old behavior, right? Even you pass all live instances, they will still be filtered out here.

Reply:
That is why we need the additional StrictMatchVerifier Builder parameter. By default it is false. So the old behavior will be kept the same.